File: sdkwork-app-sdk-typescript/src/types/vip-info-vo.ts
import type { CustomerServiceInfo } from './customer-service-info';
import type { VipBenefitVO } from './vip-benefit-vo';
import type { VipCouponVO } from './vip-coupon-vo';

/** VIP信息响应 */
export interface VipInfoVO {
  /** 创建时间 */
  createdAt?: string;
  /** 更新时间 */
  updatedAt?: string;
  /** VIP等级 */
  vipLevel?: number;
  /** VIP等级名称 */
  vipLevelName?: string;
  /** VIP状态 */
  vipStatus?: string;
  /** 生效时间 */
  effectiveTime?: string;
  /** 过期时间 */
  expireTime?: string;
  /** 剩余天数 */
  remainingDays?: number;
  /** 总天数 */
  totalDays?: number;
  /** 累计消费 */
  totalSpent?: number;
  /** 会员积分(灵点) */
  vipPoints?: number;
  /** 成长值 */
  growthValue?: number;
  /** 升级所需成长值 */
  upgradeGrowthValue?: number;
  /** 专属客服 */
  customerService?: CustomerServiceInfo;
  /** 专属权益 */
  benefits?: VipBenefitVO[];
  /** 专属优惠券 */
  coupons?: VipCouponVO[];
}

File: sdkwork-app-sdk-go/types/vip_info_vo.go
package types

// VIP信息响应
type VipInfoVO struct {
	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
	VipLevel int `json:"vipLevel"`
	VipLevelName string `json:"vipLevelName"`
	VipStatus string `json:"vipStatus"`
	EffectiveTime string `json:"effectiveTime"`
	ExpireTime string `json:"expireTime"`
	RemainingDays int `json:"remainingDays"`
	TotalDays int `json:"totalDays"`
	TotalSpent int `json:"totalSpent"`
	VipPoints int `json:"vipPoints"`
	GrowthValue int `json:"growthValue"`
	UpgradeGrowthValue int `json:"upgradeGrowthValue"`
	CustomerService CustomerServiceInfo `json:"customerService"`
	Benefits []VipBenefitVO `json:"benefits"`
	Coupons []VipCouponVO `json:"coupons"`
}
